### PR TITLE
OWNERS.md: add notifications project area

### PR DESCRIPTION
**Pull request by Rugvip**
This introduces the Notifications Project Area 🎉 

The scope is the notifications and the signals backend. The signals backend is currently closely tied to the notifications system and part of its initial implementation, although it is a standalone feature that may be used by other plugins.

@drodil and @mareklibra have both been doing excellent work towards in the early implementations in the signals and notifications plugins along with the the external implementation in https://github.com/janus-idp/backstage-plugins/tree/main/plugins/notifications that is being brought in, as well as aligning on the design in [BEP-0001](https://github.com/backstage/backstage/tree/master/beps/0001-notifications-system).

Made it an incubating area for now, which means that ownership is shared with core maintainers, who will also help out with reviews. That's ofc something that we can change down the line though.

### CODE REVIEW

**Review by drodil — CHANGES_REQUESTED** [review]
I have to double check the company policy for this

**Comment by mareklibra** [comment]
Thank you. I can stay involved in these areas. Happy to move it forward!

File: OWNERS.md [hunk 1]
```diff
@@ -107,6 +107,17 @@ Scope: The TechDocs plugin and related tooling
...
 
+### Notifications
+
+Team: @backstage/notifications-maintainers
+
+Scope: The Notifications and Signals plugins and libraries
+
+| Name        | Organization       | GitHub                                      | Discord      |
+| ----------- | ------------------ | ------------------------------------------- | ------------ |
+| Heikki      | OP Financial Group | [drodil](https://github.com/drodil)         | `deathammer` |
```
Comment (drodil) [inline OWNERS.md:118]: So got some more information and unfortunately I cannot be part of this even if the organization information is removed.

Comment (Rugvip) [reply OWNERS.md]: Alright, removed you for now and let's hope it can be worked out still 🤞

**Comment by Rugvip** [comment]
Updated to just @mareklibra for now so that we can get you write access 😁

**Review by freben — APPROVED** [review]

**Pull request merged**